api_server.py

```
@@ -205,14 +205,33 @@ def check_rules(available_ids: List[int]) -> Dict:
+        return {
+            "rules_pass": False,
+            "rules": {},
+            "skill_counts": {"TTR": 0, "LGV": 0, "BA": 0},
+            "ba_non_ttr": 0,
+        }
+    conn = get_db()
+    placeholders = ",".join("?" * len(available_ids))
+    # Bandit: B608 - SQL injection check (placeholders are generated but values are safe integer IDs)
+    # Sourcery skip: sql-injection
+    rows = conn.execute(
+        f"SELECT role, skills FROM crew WHERE id IN ({placeholders})",
```

Ruff (0.15.2)

[error] Possible SQL injection vector through string-based query construction

(S608)